troubleshooting/incrementally_configuring_images.md

```diff
@@ -0,0 +1,344 @@
+
+   ```bash
+   kubectl get workflows -n argo | grep prepare-images
+   ```
+
+   Example output:
+
+   ```bash
+   cos-products-20231031-4x8wz-prepare-images-bwgl7      Failed      2d7h
+   cos-products-20231031-gdm63-prepare-images-2rkds      Failed      2d4h
+   cos-products-20231031-v8u11-prepare-images-bgs7x      Succeeded   47h
+   ```
+
+1. (`ncn-mw#`) Find the workflow SAT logs.
+
+   ```bash
+   cray artifacts list config-data --format json | jq '.artifacts[].Key' | grep $IMAGE_WORKFLOW_ID | grep sat-wrapper
+   ```
+
+   Example output:
+
+   ```bash
+   "argo/cos-products-20231031-gdm63-prepare-images-2rkds/cos-products-20231031-gdm63-prepare-images-2rkds-sat-wrapper-2664865581/main.log"
+   "argo/cos-products-20231031-gdm63-prepare-images-2rkds/cos-products-20231031-gdm63-prepare-images-2rkds-sat-wrapper-628804191/main.log"
+   ```
+
+1. (`ncn-mw#`) Download the workflow SAT logs.
+
+   ```bash
+   cray artifacts get config-data "<log name>" main.log
+   ```
+
+1. (`ncn-mw#`) Set `BASE_IMAGE_ID`.
+
+   View the log and look for a line the includes `Creation of image`
+
+   ```bash
+   INFO: Creation of image ssi-cne-1.0.0-32-csm-1.5.x86_64-csm-1.5.0.beta37-10 succeeded: ID f5374c4c-8c5a-4b79-a5e5-aef778ed36cd
+   ```
+
+   Set `BASE_IMAGE_ID` to the value of the image id.  It is important that this value is set to the image id and not the image name.
+
+   ```bash
+   export BASE_IMAGE_ID="f5374c4c-8c5a-4b79-a5e5-aef778ed36cd"
+   ```
+
+   SAT may have generated multiple images, in which case it is up to the user to determine which is the desired base image.
+
+1. (`ncn-mw#`) Set the Argo `CONFIG_WORKFLOW_ID`.
+
+   ```bash
+   export CONFIG_WORKFLOW_ID=<workflow id>
+   ```
+
+   If the exact workflow id is unknown, look for recent `prepare-images` workflows.
+
+   ```bash
+   kubectl get workflows -n argo | grep update-cfs-config
+   ```
+
+   Example output:
+
+   ```bash
+   cos-products-20231031-54sbq-update-cfs-config-wk8g2   Succeeded   2d7h
+   cos-products-20231031-6giwt-update-cfs-config-gdv7b   Failed      2d7h
+   cos-products-20231031-e5xvg-update-cfs-config-fdv97   Succeeded   27h
+   cos-products-20231031-sm5qz-update-cfs-config-hh522   Failed      41m
+   cos-products-20231031-x0xa5-update-cfs-config-x8kwf   Failed      2d19h
+   ```
+
+1. (`ncn-mw#`) Find the workflow SAT logs.
+
+   ```bash
+   cray artifacts list config-data --format json | jq '.artifacts[].Key' | grep $CONFIG_WORKFLOW_ID | grep sat-wrapper
+   ```
+
+   Example output:
+
+   ```bash
+   "argo/cos-products-20231031-e5xvg-update-cfs-config-fdv97/cos-products-20231031-e5xvg-update-cfs-config-fdv97-sat-wrapper-3214890915/main.log"
+   "argo/cos-products-20231031-e5xvg-update-cfs-config-fdv97/cos-products-20231031-e5xvg-update-cfs-config-fdv97-sat-wrapper-3848811723/main.log"
+   ```
+
+1. (`ncn-mw#`) Download the workflow SAT logs.
+
+   ```bash
+   cray artifacts get config-data "<log name>" main.log
+   ```
```

Comment:
Suggesting a similar simplification that uses the IUF logs here as well:

````suggestion
1. (`ncn-mw#`) Find the SAT logs by looking for a line like the following in the IUF logs:

   ```text
   2023-11-03T17:31:50.121175Z DBG  [update-managed-cfs-config                                ] LOG FILE FOR sat-bootprep-run: argo/cos-products-20231031-sm5qz-update-cfs-config-hh522/cos-products-20231031-sm5qz-update-cfs-config-hh522-sat-wrapper-1367779725/main.log
   ```

   Note that this may have been logged in a different session within the IUF activity, so a
   different IUF log directory may need to be examined.

   The string starting with "argo/" is the key within the `config-data` bucket of the S3 object
   which contains the log output from `sat bootprep`. Save this S3 key to an environment variable:

   ```bash
   export LOGS_S3_KEY="argo/cos-products-20231031-gdm63-prepare-images-2rkds/cos-products-20231031-gdm63-prepare-images-2rkds-sat-wrapper-2664865581/main.log"
   ```

1. (`ncn-mw#`) Download the workflow SAT logs.

   ```bash
   cray artifacts get config-data "$LOGS_S3_KEY" main.log
   ```
````